make/autoconf/flags-cflags.m4

```diff
@@ -945,39 +945,60 @@ AC_DEFUN([FLAGS_SETUP_CFLAGS_CPU_DEP],
       AC_LANG_PUSH(C)
       OLD_CFLAGS="$CFLAGS"
       CFLAGS="$CFLAGS -march=armv8-a+sve"
+      # check the compiler and binutils support sve or not
       AC_MSG_CHECKING([if Arm SVE ACLE is supported])
-      AC_COMPILE_IFELSE([AC_LANG_PROGRAM([#include <arm_sve.h>],
+      AC_COMPILE_IFELSE([AC_LANG_PROGRAM(
           [
-            svint32_t r = svdup_n_s32(1);
-            return 0;
-          ])],
-          [
-            AC_MSG_RESULT([yes])
-            $2SVE_CFLAGS="-march=armv8-a+sve"
-            # Switching the initialization mode with gcc from 'pattern' to 'zero'
-            # avoids the use of unsupported `__builtin_clear_padding` for variable
-            # length aggregates
-            if test "x$DEBUG_LEVEL" != xrelease && test "x$TOOLCHAIN_TYPE" = xgcc ; then
-              INIT_ZERO_FLAG="-ftrivial-auto-var-init=zero"
-              FLAGS_COMPILER_CHECK_ARGUMENTS(ARGUMENT: [$INIT_ZERO_FLAG],
-                IF_TRUE: [
-                  $2SVE_CFLAGS="${$2SVE_CFLAGS} $INIT_ZERO_FLAG"
-                ]
-              )
-            fi
+            #include <arm_sve.h>
+            svfloat64_t a() {}
           ],
           [
-            AC_MSG_RESULT([no])
-            $2SVE_CFLAGS=""
-          ]
+            svint32_t r = svdup_n_s32(1)
+          ])],
+          [sve_supported=yes],
+          [sve_supported=no]
       )
+      AC_MSG_RESULT([$sve_supported])
+      $2SVE_CFLAGS=""
+      if test "x$enable_aarch64_sve" = "xyes"; then
```

Comment:
`UTIL_ARG_ENABLE` will return `true`, not `yes`. Have you even tested this code?

Reply:
I did the test before. I was also confused that the `enable_aarch64_sve` sometimes return `yes`/`no`, sometimes return `true`/`false`, it seems depends the code implementations.

The log snippet after run bash configure.

```shell
> grep "enable_aarch64_sve=" build/linux-aarch64-server-release/configure.log -n
1117:+++ eval 'enable_aarch64_sve=$ac_optarg'
1118:++++ enable_aarch64_sve=yes
20015:+++ enable_aarch64_sve=true
```

[configure.log](https://github.com/user-attachments/files/22246010/configure.log)

Reply:
So if you used the AC_ version, it will set enable_aarch64_sve with yes/no, but then you switched to the UTIL_ version, but you kept the same name as argument. That just causes a mess. I guess the UTIL_ version, which calls the AC_function at its core, will override the value. But just don't use that name.